addons/website_slides/controllers/main.py

```diff
@@ -1117,7 +1117,11 @@ def slides_embed(self, slide_id, page="1", **kw):
         # determine if it is embedded from external web page
         referrer_url = request.httprequest.headers.get('Referer', '')
         base_url = request.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        is_embedded = referrer_url and not bool(base_url in referrer_url) or False
+        websites = request.env['website'].search([])
+        is_embedded = False
+        if referrer_url:
+            web_domains = [wb.domain for wb in websites if wb.domain] + [base_url, request.httprequest.host]
```

Comment:
if you embed slide from website 1 in website 2... it is embedded... No sense / wrong to check in other websites...
So you should use slide.get_base_url() (from line 1127) 

But, in which case request.httprequest.host is != referral and that you will be not in embed ? :thinking:

Reply:
That is the problem with the detection, there can be multiple websites, for example we can access the slides with 127.0.0.1 or localhost (but not only locally, you can also have multiple domains pointing to the same webserver), so the detection must look through all the possible websites,

but we are working on this, as it is quite a big bug to fix, so it was added to the RD SM pipeline here https://www.odoo.com/web#id=2618753&action=333&active_id=965&model=project.task&view_type=form&cids=1&menu_id=4720 if you want to see their take on the bug to fix

Nice idea to use the slide variable at 1127, I haven't thought of that, I'll investigate :smile:

Reply:
>  you can also have multiple domains pointing to the same webserver

Yes and with 127.0.0.2 and 127.0.0.3 and 127.0.0.4 ... ;)

Wrong with odoo, each website have an unique domain, you cannot configure several domains.
And if you have wrongly configured your server, Odoo will serve the first website (-> not a bug, wrong configuration)
Useful in test environment not in prod where your canonical urls will be wrong too.

But since it is a big bug :thinking: , it is for RD SM pipelines... my bad :1st_place_medal:

Reply:
> Wrong with odoo, each website have an unique domain, you cannot configure several domains.
> And if you have wrongly configured your server, Odoo will serve the first website (-> not a bug, wrong configuration)
> Useful in test environment not in prod where your canonical urls will be wrong too.

But what about Website > Configuration > Websites (with debug mode) where you can register multiple website each with a different domain, all pointing to the same database if I understood correctly ? so if the slides are accessed via website 1 or website 2, we will need to check both (web.base.url of the env should point to the correct url so that's why it was used before but if we simply replace one domain with another one in the url and refresh the page, web.page.url isn't refreshed correctly, and there are probably other cases where this can happen)

> But since it is a big bug , it is for RD SM pipelines... my bad

No problem ! Any help is appreciated on how to fix this the most correct way possible 😄

Reply:
>  you can register multiple website each with a different domain, all pointing to the same database if I understood correctly

Yes it is true.
But as well your slide is visible on One website, as well on all Websites.
But in a well configured db, you are embed as soons as your url_root is not the same that your referer... no ? 
if you correctly show your slide with slide.get_base_url().... ofc. If it is not the case you already have a lot of other bug in term of seo.
And imo a slide X with website_id = 1, should be considered as embedded if you show it on website 2.

Reply:
That is how it is working right now in the code but the embedding is mostly there when you are showing the slide on a "third-party website" let's say, so for example https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_iframe, and not inside your odoo environment from what I understood, which is why this needs to be fixed

I'll have to look deeper into the characteristics of the url, url_host, url_root, and referrer to see what shoud be best to compare 

Thanks for your input, it's greatly appreciated 🙏